### PR TITLE
TE: Enabling Application start

### PR DESCRIPTION
**Pull request by puneetjaiswal**
TE: Enabling Application start
- previous code had missing hibernate bundle.

### CODE REVIEW

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/db/dao/AnomalyResultDAO.java [hunk 1]
```diff
@@ -8,39 +8,39 @@
...
-      "SELECT r FROM AnomalyResult r WHERE r.collection = :collection AND r.metric = :metric "
+      "SELECT r FROM AnomalyResult r WHERE r.function.collection = :collection AND r.function.metric = :metric "
           + "AND ((r.startTimeUtc >= :startTimeUtc AND r.startTimeUtc <= :endTimeUtc) "
           + "OR (r.endTimeUtc >= :startTimeUtc AND r.endTimeUtc <= :endTimeUtc))";
 
   private static final String FIND_BY_COLLECTION_TIME_METRIC_DIMENSION =
-      "SELECT r FROM AnomalyResult r WHERE r.collection = :collection" + "AND r.metric = :metric "
-          + "AND r.dimensions IN :dimensions "
+      "SELECT r FROM AnomalyResult r WHERE r.function.collection = :collection" + "AND r.function.metric = :metric "
+          + "AND r.function.dimensions IN :dimensions "
```
Comment (npawar) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/db/dao/AnomalyResultDAO.java]: AnomalyFunction.dimension is not the same as AnomalyResults.dimensions. In anomaly function, dimensions stores the dimensions to use for drill down (eg: countryCode), whereas in anomaly results, this value is the actual dimension key (eg: (*,*,*,norway,*)

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/db/entity/AnomalyResult.java [hunk 2]
```diff
@@ -6,44 +6,25 @@
...
-  @Column(name = "collection", nullable = false)
-  private String collection;
-
   @Column(name = "start_time_utc", nullable = false)
   private Long startTimeUtc;
 
   @Column(name = "end_time_utc", nullable = true)
   private Long endTimeUtc;
 
-  @Column(name = "dimensions", nullable = false)
```
Comment (npawar) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/db/entity/AnomalyResult.java]: Need to retain dimensions. Rename it if this is being confused between the dimensions in anomaly functions

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/detector/function/MinMaxThresholdFunction.java [hunk 3]
```diff
@@ -75,10 +75,6 @@
 
       if (deviationFromThreshold != 0) {
         AnomalyResult anomalyResult = new AnomalyResult();
-        anomalyResult.setCollection(getSpec().getCollection());
-        anomalyResult.setMetric(metric);
-        anomalyResult.setDimensions(CSV.join(dimensionKey.getDimensionValues()));
```
Comment (npawar) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/detector/function/MinMaxThresholdFunction.java]: setDimensions is needed here

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/detector/function/UserRuleAnomalyFunction.java [hunk 4]
```diff
@@ -80,16 +82,10 @@ private AnomalyResult getMergedAnomalyResults(List<AnomalyResult> anomalyResults
     AnomalyResult firstAnomalyResult = anomalyResults.get(0);
     AnomalyResult lastAnomalyResult = anomalyResults.get(n - 1);
     AnomalyResult mergedAnomalyResult = new AnomalyResult();
-    mergedAnomalyResult.setCollection(firstAnomalyResult.getCollection());
-    mergedAnomalyResult.setMetric(firstAnomalyResult.getMetric());
-    mergedAnomalyResult.setDimensions(firstAnomalyResult.getDimensions());
```
Comment (npawar) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/detector/function/UserRuleAnomalyFunction.java]: same for setDimensions as above

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/db/dao/AnomalyResultDAO.java (hunk 1 above)
Comment (puneetjaiswal) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/db/dao/AnomalyResultDAO.java]: OK.

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/db/entity/AnomalyResult.java (hunk 2 above)
Comment (puneetjaiswal) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/db/entity/AnomalyResult.java]: done.

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/detector/function/MinMaxThresholdFunction.java (hunk 3 above)
Comment (puneetjaiswal) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/detector/function/MinMaxThresholdFunction.java]: restored dimensions.

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/detector/function/UserRuleAnomalyFunction.java (hunk 4 above)
Comment (puneetjaiswal) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/detector/function/UserRuleAnomalyFunction.java]: restored dimensions.

**Comment by npawar** [comment]
LGTM

**Pull request merged**